### PR TITLE
Feat/functionality

### LINKED ISSUE #13: Build Poster component & style

### LINKED ISSUE #16: As a user, I should be able to fill out the form on the create page and click submit, which will take me to the poster page corresponding with and displaying the quote I made

### LINKED ISSUE #18: As a user I should be able to click on the chaotic quote generator button and be taken to a page where I can view a randomly generated card with a chaotic quote and a button to generate a new chaotic quote

### LINKED ISSUE #19: As a user I should be able to favorite a randomly generated quote

### LINKED ISSUE #21: As a user I should be able to click the generate quote button on the wholesome page and see a new randomly generated wholesome quote

### LINKED ISSUE #22: As a user I should be able to click the generate quote button on the chaotic page and see a new randomly generated chaotic quote

### LINKED ISSUE #26: As a user, when I click the logo I go to the home page

### LINKED ISSUE #28: Add button to Poster that generates quote of the opposite type/navigates

### PR DESCRIPTION
**Pull request by rachelsoae**
What I did:
- Generate a random poster of the correct type when that type's Generate button is clicked
- Refactor Favorites and Card to allow the same Card component to be used as the Poster, with dynamic styling
- Dynamically change color of the Poster page based on category
- Add buttons to Poster page that allow user to favorite and generate random posters of either type
- Draft GETs for page load

I think that's it....

Resolves #13 
Resolves #16 
Resolves #18 
Resolves #19 
Resolves #21 
Resolves #22 
Resolves #26 
Resolves #28 

Did this break anything?
- [x] No
- [ ] Yes

Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Super small fix (Corrected a typo, removed a comment, etc.)
- [ ] Skip all the other stuff and briefly explain the fix.

Checklist:
- [ ] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [x] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

What's next?
Finish BE and then refactor FE to integrate API data in place of mock data.
Mock data and all references to it can be deleted once real data is working.

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -4,10 +4,75 @@ import Favorites from '../Favorites/Favorites.js'
...
+import Poster from '../Poster/Poster.js'
+import { mockQuotes, mockImages, mockPosters } from '../mockData'
+import {useState, useEffect } from 'react'
+import { getData } from '../apiCalls'
 
 const App = () => {
+  const [quotes, setQuotes] = useState(mockQuotes.quotes)
+  const [images, setImages] = useState(mockImages.images)
+
+  //// --> hopefully ready for real data - needs testing/debugging
```
Comment (AlexaVanSchaardenburg) [inline src/App/App.js:16]: So excited that these are here! I am hoping once the be is up and running it is smooth sailing!

File: src/App/App.js [hunk 2]
```diff
@@ -4,10 +4,75 @@ import Favorites from '../Favorites/Favorites.js'
...
+      type: type
+    })
+  }
+
+  const [fontSize, setFontSize] = useState('');
+
+  const changeFontSize = (param) => {
+    setFontSize(param === 'poster' ? '2.5em' : '1em')
+  }
+
```
Comment (AlexaVanSchaardenburg) [inline src/App/App.js:55]: I would love to have this syntax explained to me, because I can not wrap my head around it

File: src/App/App.js [hunk 3]
```diff
@@ -16,15 +81,15 @@ const App = () => {
...
+      <Nav changeFontSize={changeFontSize} />
       <Routes>
-        <Route path='/' element={<Home />}/>
-        <Route path='favorites' element={<Favorites />}/>
-        <Route path='create' element={<Form />}/>
-        <Route path='/random-quote-:quote-type' element={<Favorites />}/>
+        <Route path='/' element={<Home randomizePoster={randomizePoster} changeFontSize={changeFontSize} />}/>
+        <Route path='/favorites' element={<Favorites />}/>
+        <Route path='/create' element={<Form setPoster={setPoster} />}/>
+        <Route path='/:type' element={<Poster poster={poster} font={fontSize} saveFavorite={saveFavorite} randomizePoster={randomizePoster} />}/>
```
Comment (AlexaVanSchaardenburg) [inline src/App/App.js:90]: I like this path, very smooth and readable

File: src/Card/Card.js [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
-        <img src='https://www.mattpaynephotography.com/images/960/San-Juan-Mountains-Sunrise-Panorama.webp' id='quote-img'/>
-        <p id='quote' >“Courage is more exhilarating than fear and in the long run it is easier. We do not have to become heroes overnight. Just one step at a time, meeting each thing that comes up, seeing it is not as dreadful as it appeared, discovering we have the strength to stare it down.”</p>
+    <article className='frame'>
+      <div 
+        className='img' 
+        style={{
+          backgroundImage: `url(${image})`,
+          backgroundPosition: 'center',
+          backgroundSize: 'cover'
+        }}
```
Comment (AlexaVanSchaardenburg) [inline src/Card/Card.js:14]: I played around with this inline styling, but could not get it to look nice, I am glad you got it!

File: src/Form/Form.js [hunk 5]
```diff
@@ -1,38 +1,92 @@
...
+    <main id='form-page' >
       <h2>make your own quote!</h2>
-      <p>Your quote poster will be added to the quotes and can be generated randomly by future users.</p>
+      {/* <p>Your quote poster will be added to the quotes and can be generated randomly by future users.</p> */}
       <form onSubmit={handleSubmit}>
         <label htmlFor='type-of-poster'>1. select your quote type</label>
         <div className='buttons'>
-          <button className='form-button' id='wholesome-button'>🥹 wholesome quote</button>
-          <button className='form-button' id='chaotic-button'>😈 chaotic quote</button>
+          <label htmlFor="wholesome-radio" className='form-button' id='wholesome-button'>
```
Comment (AlexaVanSchaardenburg) [inline src/Form/Form.js:37]: Why is it important to use radio buttons? I just don't love the styling and we can tie there values to the same state to get the back and forth effect - then add a little of our own styling to indicate which button is clicked?

File: src/Nav/Nav.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
 import './Nav.css';
-import { NavLink } from 'react-router-dom'
+import { NavLink, Link } from 'react-router-dom'
 
 const Nav = () => {
   return (
     <nav>
-      <h1 id='nav-title'>humor me</h1>
+      <Link to='/'><h1 id='nav-title'>humor me</h1></Link>
```
Comment (AlexaVanSchaardenburg) [inline src/Nav/Nav.js:7]: Always love adding this detail by making the 'logo' clickable to bring the user to home

**Review by AlexaVanSchaardenburg — COMMENTED** [review]
This looks great! Below are my comments:
- the code is so clean and DRY and I LOVE it
- Form works great! Well done! 
- We need to refactor the card component to get rid of the trash button, and then decide how we want to display that on the favorites page
- When we hover over the emoji 'buttons' on the poster they have the clicking pointer - but they aren't clickable  we will need to make sure to refactor that once all of the buttons are straightened out
I have also left code specific comments! Let me know if you have any questions!

File: src/Favorites/Favorites.js [hunk 7]
```diff
@@ -1,22 +1,14 @@
 import './Favorites.css';
 import Card from '../Card/Card.js'
+import { mockPosters } from '../mockData'
 
 const Favorites = () => {
-
-  //eventually we will bring the favorites in from the app, map through them, and then display them on the dom using the Cards component
+  const favorites = mockPosters.posters.map(poster => <Card key={poster.id} image={poster.src} quote={poster.quote.text} type={poster.quote.type} font='1.3em'/>)
```
Comment (dcochico) [inline src/Favorites/Favorites.js:6]: nice job DRY'ing this section up

File: src/Poster/Poster.js
Comment (dcochico) [inline src/Poster/Poster.js:1]: once we integrate our API, we will need to call a useEffect hook to still allow poster randomization on page refresh

**Review by dcochico — APPROVED** [review]
great work, and i've left some notes for you on a few of the files.

**Pull request merged**